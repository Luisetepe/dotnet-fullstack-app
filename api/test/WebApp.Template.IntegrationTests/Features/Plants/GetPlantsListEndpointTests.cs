using Microsoft.EntityFrameworkCore;
using Snapshooter;
using Snapshooter.Xunit;
using WebApp.Template.Api.Endpoints.Plants;
using WebApp.Template.Application.Features.Plants.Queries.GetPlantsList;

namespace WebApp.Template.IntegrationTests.Features.Plants;

public class GetPlantsListEndpointTests(IntegrationTestFixture fixture)
    : TestBase<IntegrationTestFixture>
{
    [Theory]
    [InlineData(1, 5, "name", "asc")]
    [InlineData(1, 5, "name", "desc")]
    [InlineData(1, 5, "status", "asc")]
    [InlineData(1, 5, "status", "desc")]
    public async Task Should_Return_Plants(
        int page,
        int pageSize,
        string sortBy,
        string sortDirection
    )
    {
        // Act
        var (httpResponse, result) = await fixture.Client.GETAsync<
            GetPlantsListEndpoint,
            GetPlantsListRequest,
            GetPlantsListResponse
        >(
            new GetPlantsListRequest
            {
                PageNumber = page,
                PageSize = pageSize,
                OrderBy = sortBy,
                Order = sortDirection
            }
        );

        // Assert
        httpResponse.StatusCode.Should().Be(HttpStatusCode.OK);

        result.Should().NotBeNull();
        result
            .Should()
            .MatchSnapshot(
                SnapshotNameExtension.Create(page, pageSize, sortBy, sortDirection),
                matchOptions => matchOptions.IgnoreFields("**.Id")
            );
    }
}
