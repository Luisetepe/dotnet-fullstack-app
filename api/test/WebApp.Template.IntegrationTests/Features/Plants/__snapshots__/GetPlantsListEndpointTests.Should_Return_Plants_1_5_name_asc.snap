{
  "Plants": [
    {
      "Id": "0G5GJEQ9WF6YJ",
      "Name": "Plant 1",
      "PlantId": "PL01",
      "UtilityCompany": "Company 1",
      "Status": "Operational",
      "Tags": [
        "TAG01",
        "TAG02"
      ],
      "CapacityDc": 100.0,
      "Portfolios": [
        "Portfolio 1",
        "Portfolio 2"
      ]
    },
    {
      "Id": "0G5GJEQ9WF6YK",
      "Name": "Plant 2",
      "PlantId": "PL02",
      "UtilityCompany": "Company 2",
      "Status": "Under construction",
      "Tags": [
        "TAG02",
        "TAG03"
      ],
      "CapacityDc": 200.0,
      "Portfolios": [
        "Portfolio 2",
        "Portfolio 3"
      ]
    },
    {
      "Id": "0G5GJEQ9WF6YM",
      "Name": "Plant 3",
      "PlantId": "PL03",
      "UtilityCompany": "Company 3",
      "Status": "Decommissioned",
      "Tags": [
        "TAG03",
        "TAG04"
      ],
      "CapacityDc": 200.0,
      "Portfolios": [
        "Portfolio 3",
        "Portfolio 4"
      ]
    },
    {
      "Id": "0G5GJEQ9WF6YN",
      "Name": "Plant 4",
      "PlantId": "PL04",
      "UtilityCompany": "Company 4",
      "Status": "Planned",
      "Tags": [
        "TAG04",
        "TAG05"
      ],
      "CapacityDc": 300.0,
      "Portfolios": [
        "Portfolio 4",
        "Portfolio 5"
      ]
    },
    {
      "Id": "0G5GJEQ9WF6YP",
      "Name": "Plant 5",
      "PlantId": "PL05",
      "UtilityCompany": "Company 5",
      "Status": "Closed",
      "Tags": [
        "TAG05",
        "TAG01"
      ],
      "CapacityDc": 400.0,
      "Portfolios": [
        "Portfolio 5"
      ]
    }
  ],
  "Pagination": {
    "CurrentPageNumber": 1,
    "CurrentPageSize": 5,
    "TotalRows": 5,
    "TotalPages": 1
  }
}
